### PR TITLE
Remove ledger_commit_info and flatten QC structure

### PR DESCRIPTION
**Pull request by ariqchowdhury**
resolves category-labs/category-internal#1009, resolves category-labs/category-internal#1011, category-labs/category-internal#1012

### CODE REVIEW

File: monad-state/src/statesync.rs [hunk 1]
```diff
@@ -127,10 +127,10 @@ where
             self.proposal_buffer.pop_front();
         }
 
-        if commit_condition(block_header.round, block_header.qc.info) {
+        if block_header.qc.is_commitable() {
             let committed_seq_num = block_header.seq_num - self.commit_distance;
```
Comment (omegablitz) [inline monad-state/src/statesync.rs]: committed_seq_num should be block_header.qc.get_parent_seq_num()

block_header.seq_num isn't trusted

File: monad-proto/proto/ledger.proto
Comment (omegablitz) [inline monad-proto/proto/ledger.proto:1]: I guess this file can be deleted, doesn't really matter though

Comment (ariqchowdhury) [reply monad-proto/proto/ledger.proto:1]: yeah, all the proto stuff will be removed at some point

File: monad-state/src/statesync.rs (hunk 1 above)
Comment (ariqchowdhury) [reply monad-state/src/statesync.rs]: are you saying the original code had an issue with using block_header.seq_num?

Comment (omegablitz) [reply monad-state/src/statesync.rs]: Yes the original code is messed up also

File: monad-state/src/statesync.rs [hunk 2]
```diff
@@ -128,7 +128,14 @@ where
...
         if block_header.qc.is_commitable() {
-            let committed_seq_num = block_header.seq_num - self.commit_distance;
+            let latest_seq_num =
+                if let Some(b) = self.full_blocks.get(&block_header.get_parent_id()) {
+                    b.get_seq_num()
+                } else {
+                    return None;
+                };
+
+            let committed_seq_num = latest_seq_num - self.commit_distance;
```
Comment (omegablitz) [inline monad-state/src/statesync.rs:138]: I think now committed_seq_num is always latest_seq_num - SeqNum(1), but there's no meaningful difference, just that the resync_threshold will be off by one.

**Pull request merged**